crates/omnix-health/src/check/nix_version.rs

```diff
@@ -0,0 +1,63 @@
+use nix_rs::version_spec::NixVersionReq;
+
+use nix_rs::info;
+use serde::{Deserialize, Serialize};
+
+use crate::traits::*;
+
+/// Check that [nix_rs::version::NixVersion] is set to a good value.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(default, rename_all = "kebab-case")]
+pub struct NixVersionCheck {
+    #[serde(deserialize_with = "deserialize_version_req")]
+    pub supported: NixVersionReq,
+}
+
+fn deserialize_version_req<'de, D>(deserializer: D) -> Result<NixVersionReq, D::Error>
```

Comment:
This?

https://docs.rs/serde_with/latest/serde_with/derive.DeserializeFromStr.html